### ISSUE TITLE
[BUG] srGelmanRubin stopping rule doesn't stop run

### ISSUE THREAD

**Issue by ms609** [revbayes/revbayes#543]
**Describe the bug**
Convergence diagnostic stopping rules seem not to be working

**To Reproduce**

Run the examples from the stopping rules documentation pages.

```rb
# Binomial example: estimate success probability given 7 successes out of 20 trials
r ~ dnExp(10)
p := Probability(ifelse(r < 1, r, 1))
n <- 20
k ~ dnBinomial(n, p)
k.clamp(7)
mymodel = model(k)

moves = VectorMoves()
moves.append( mvSlide(r, delta=0.1, weight=1) )

paramFile = "output/parameters.log"

monitors = VectorMonitors()
monitors.append( mnModel(filename = paramFile, printgen = 100, p) )
monitors.append( mnScreen(printgen = 1000, p) )

# Stop when the potential scale reduction factor falls below 1.01
stopping_rules[1] = srGelmanRubin(1.5, file = paramFile, freq = 1000)
stopping_rules[2] = srMinESS(15, file = paramFile, freq = 1000)

# Create the MCMC object
mymcmc = mcmc(mymodel, monitors, moves, nruns = 2)

# Begin the MCMC run
mymcmc.run(rules = stopping_rules)
```

**Expected behavior**
The run should stop pretty promptly – an ESS of 15 is trivial, as is a PSRF of 1.5.

But it continues almost indefinitely; Tracer confirms an ESS > 2500.

When I set `stopping_rules[1] = srGelmanRubin(5, file = paramFile, freq = 1000)`, the run **did** stop at iteration 1000.

Hypothesis: Are the convergence diagnostics only being triggered at iteration `freq`, rather than every `freq` iterations?

**Computer info**
Win 10, rb1.2.5

**Comment by davidcerny** [comment]
I can't reproduce this. For me, it's specifically `srGelmanRubin` that's problematic. If I only use `srMinESS`, the analysis indeed stops after 2000–3000 iterations. This is despite the fact that when I run the resulting log files through coda's `gelman.diag()`, I get:

```
Likelihood      Prior          p          r 
  1.000743   1.000559   1.000559   1.000559
```

> Hypothesis: Are the convergence diagnostics only being triggered at iteration `freq`, rather than every `freq` iterations?

I'm pretty sure that's not the case. `GelmanRubinStoppingRule` inherits from `AbstractConvergenceStoppingRule`, which defines a `checkAtIteration` method just the way we'd expect. I'm wondering if there's something wrong with our calculation of the Gelman-Rubin statistic itself.

Also, these problems would be so much easier to diagnose if the specified stopping rules were automatically added to the `mnScreen` (and perhaps also `mnFile`) monitor...

**Comment by bjoelle** [comment]
so if I understand this right, the Gelman-Rubin is supposed to be calculated based on the same _variable_ in different _chains_ - but the current code appears to be doing the reverse (see `GelmanRubinStoppingRule::stop` - we only load one trace at a time) so we compare different _variables_ within the same _chain_
possibly that's the source of the issue ?

**Comment by ms609** [comment]
It sounds like you've put your finger on the issue, @bjoelle; as you say, we should be comparing the estimates of each variable between the cold chains of each run.

**Comment by bredelings** [comment]
:facepalm:

**Comment by davidcerny** [comment]
Fixed by #555.

**Comment by davidcerny** [comment]
Fixed by #555.
